opal/mca/threads/base/base.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2014 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2014 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ *
+ */
+
+#ifndef OPAL_THREADS_BASE_H
+#define OPAL_THREADS_BASE_H
+
+#include "opal_config.h"
+#include "opal/mca/base/mca_base_framework.h"
+#include "opal/mca/threads/threads.h"
+
+
+/*
+ * Global functions for MCA overall threads open and close
+ */
+
+BEGIN_C_DECLS
+
+/**
+ * Framework structure declaration
+ */
+OPAL_DECLSPEC extern mca_base_framework_t opal_threads_base_framework;
+
+END_C_DECLS
+
+/* include implementation to call */
+#include MCA_threads_IMPLEMENTATION_HEADER
```

Comment:
@jsquyres At least for the initial PR we went with the static approach because we wanted the common case, Pthreads, to suffer no performance regressions at all. As the package matures, seeing how much of a performance hit we take from moving from inlines to shared library calls is definitely something we'd like to explore. This PR is more about "doing no harm" initially.